library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/processor/MutationResolverTest.kt

```diff
@@ -806,6 +806,463 @@ internal class MutationResolverTest {
 
     // endregion
 
+    // region ImageWireframe update mutations
+
+    @Test
+    fun `M identify the updated wireframes W resolveMutations {Image, position}`(forge: Forge) {
+        // Given
+        val fakePrevSnapshot = forge.aList(size = forge.anInt(min = 2, max = 10)) {
+            forge.getForgery(MobileSegment.Wireframe.ImageWireframe::class.java)
+        }
+        val fakeUpdateSize = forge.anInt(min = 1, max = fakePrevSnapshot.size)
+        val fakeUpdatedWireframes = fakePrevSnapshot
+            .take(fakeUpdateSize)
+            .map { it.copy(x = forge.aLong(), y = forge.aLong()) }
+        val fakeCurrentSnapshot = fakeUpdatedWireframes + fakePrevSnapshot.drop(fakeUpdateSize)
+        val expectedUpdates = fakeUpdatedWireframes.map {
+            MobileSegment.WireframeUpdateMutation.ImageWireframeUpdate(
+                id = it.id(),
+                x = it.x,
+                y = it.y
+            )
+        }
+
+        // When
+        val mutations = testedMutationResolver.resolveMutations(
+            fakePrevSnapshot,
+            fakeCurrentSnapshot
+        )
+
+        // Then
+        assertThat(mutations?.adds).isNullOrEmpty()
+        assertThat(mutations?.removes).isNullOrEmpty()
+        assertThat(mutations?.updates).isEqualTo(expectedUpdates)
+    }
```

Comment:
Nice job, it seems a lot of your tests here share the same structure, maybe there could be a helper method to share the boilerplate stuff and make the tests more readable :

```kotlin
fun <T, U> verifyMutationUpdates(
    forge: Forge,
    previousSnapshotConfiguration: (T) -> T = { it },
    currentSnapshotMutation: (T) -> T = { it },
    expectedUpdate: (T) -> U
){
    // Given
    val previousSnapshot = forge.aList(size = forge.aTinyInt()) {
        previousSnapshotConfiguration(forge.getForgery<T>())
    }
    val fakeUpdateSize = forge.anInt(min = 1, max = fakePrevSnapshot.size)
    val fakeUpdatedWireframes = fakePrevSnapshot
            .take(fakeUpdateSize)
            .map { currentSnapshotMutation(it) }
    val expectedUpdates = fakeUpdatedWireframes.map { expectedUpdate(it) }

    // When
    val mutations = testedMutationResolver.resolveMutations(
        fakePrevSnapshot,
        fakeCurrentSnapshot
    )
    // Then
    assertThat(mutations?.adds).isNullOrEmpty()
    assertThat(mutations?.removes).isNullOrEmpty()
    assertThat(mutations?.updates).isEqualTo(expectedUpdates)
}

@Test
fun `M identify the updated wireframes W resolveMutations {Image, position}`(forge: Forge) {
    verifyMutationUpdates<ImageWireframe,ImageWireframeUpdate> (
        forge, 
        currentSnapshotMutation = { it.copy(x = forge.aLong(), y = forge.aLong()) }
        expectedUpdate = {
            MobileSegment.WireframeUpdateMutation.ImageWireframeUpdate(
                id = it.id(),
                x = it.x,
                y = it.y
            )
        }
    )
}

```

Reply:
yeah I am thinking to use a parameterised test maybe